crates/ty_python_semantic/src/types/enums.rs

```diff
@@ -1,19 +1,58 @@
-use rustc_hash::FxHashSet;
+use ruff_python_ast::name::Name;
+use rustc_hash::FxHashMap;
 
 use crate::{
     Db,
-    place::{Place, place_from_bindings, place_from_declarations},
+    place::{Place, PlaceAndQualifiers, place_from_bindings, place_from_declarations},
     semantic_index::{place_table, use_def_map},
-    types::{ClassLiteral, KnownClass, MemberLookupPolicy, Type},
+    types::{ClassLiteral, DynamicType, KnownClass, MemberLookupPolicy, Type, TypeQualifiers},
 };
 
+#[derive(Debug, PartialEq, Eq)]
+pub(crate) struct EnumMetadata {
+    pub(crate) members: Vec<Name>,
+    pub(crate) aliases: FxHashMap<Name, Name>,
+}
+
+impl EnumMetadata {
+    fn empty() -> Self {
+        EnumMetadata {
+            members: Vec::new(),
```

Comment:
nit: we could enforce immutability here and maybe save a tiny bit of memory

```suggestion
    pub(crate) members: Box<[Name]>,
    pub(crate) aliases: FxHashMap<Name, Name>,
}

impl EnumMetadata {
    fn empty() -> Self {
        EnumMetadata {
            members: Box::default(),
```